odoo/addons/base/models/res_config.py

```diff
@@ -472,8 +467,7 @@ def _get_classified_fields(self):
             elif name.startswith('module_'):
                 if field.type not in ('boolean', 'selection'):
                     raise Exception("Field %s must have type 'boolean' or 'selection'" % field)
-                module = IrModule.sudo().search([('name', '=', name[7:])], limit=1)
-                modules.append((name, module))
+                modules += IrModule.sudo().search([('name', '=', name[7:])], limit=1)
```

Comment:
We could get the same effect by collecting all module names, search for them once, and then rebuild the list as before.  Something like:
```py
module_names = []
for name, field in self._fields.items():
    ...
    elif name.startswith('module_'):
        ...
        module_names.append(name[7:])
    ...

name_modules = {
    module.name: module
    for module in IrModule.sudo().search([('name', 'in', module_names)])
}
modules = [
    ('module_' + name, name_modules.get(name, IrModule)
    for name in module_names
]
```
and everything else remains the same.  All the records that appear in the list `modules` come from the same recordset, so they will be read as a single batch.